src/orchestration.d.ts

```diff
@@ -0,0 +1,311 @@
+
+// marker interface
+interface QueryResult {}
+
+/**
+ * An object for access the core functions of a remote chain.
+ */
+export interface Chain {
+  getChainInfo: () => Promise<ChainInfo>;
+  /**
+   * Provide (get or make) an account on the chain. The account is a
+   * named account associated with the current orchestrator instance
+   * (typically associated with a specific seat). If an account for this `Chain`
+   * with the provided `petName` already exists, it is returned,
+   * otherwise a new account is created on the remote Chain.
+   * @param petName
+   * @returns the account that controls the
+   */
+  provideAccount: (petName?: string) => Promise<OrchestrationAccount>;
+  /* query external chain state */
+  query: (queries: Proto3JSONMsg[]) => Promise<Iterable<QueryResult>>;
+
+  // TODO we need a way to have multiple offers get the same orchestrator.
+}
+
+/**
+ * An object that supports low-level queries and operations for an account on a remote chain.
+ */
+export interface ChainAccount {
+  /**
+   * @returns the address of the account on the chain
+   */
+  getAddress: () => ChainAddress;
+  /**
+   * Submit a transaction on behalf of the remote accoutn for execution on teh remote chain.
+   * @param msgs - records for the transaction
+   * @returns void
+   */
+  executeTx: (msgs: Proto3JSONMsg[]) => Promise<void>;
+  /**
+   * Submit a transaction on behalf of the remote accoutn for execution on teh remote chain.
+   * @param msgs - records for the transaction
+   * @returns void
+   */
+  executeEncodedTx: (msgs: EncodeObject[]) => Promise<void>;
+  /** deposit payment from zoe to the account*/
+  deposit: (payment: Payment) => Promise<void>;
+  /** get Purse for a brand to .withdraw() a Payment from the account */
+  getPurse: (brand: Brand) => Promise<Purse>;
+  /**
+   * Close the remote account
+   */
+  close: () => Promise<void>;
+  /* transfer account to new holder */
+  prepareTransfer: () => Promise<Invitation>;
+}
+
+export type BrandOrDenom = Brand | Denom;
+
+/**
+ * An object that supports high-level operations for an account on a remote chain.
+ */
+export interface OrchestrationAccount {
+  /** @returns the underlying low-level operation object. */
+  getChainAcccount: () => Promise<ChainAccount>;
+  /** @returns an array of amounts for every balance in the account. */
+  getBalances: () => Promise<Amount[]>;
+  /** @returns the balance of a specific denom for the account. */
+  getBalance: (denom: BrandOrDenom) => Promise<Amount | undefined>;
+
+  getDenomTrace: (
+    denom: string,
+  ) => Promise<{ path: string; base_denom: string }>;
+  /**
+   * @returns all active delegations from the account to any validator (or [] if none)
+   */
+  getDelegations: () => Promise<Delegation[]>;
+  /**
+   * @returns the active delegations from the account to a specific validator (or [] if none)
+   */
+  getDelegation: (validator: ValidatorAddress) => Promise<Delegation[]>;
```

Comment:
_question_

Can there be multiple delegations to a specific validator? This function feels more to me `getDelegationsForValidator`.

Also, should `Delegation` type be `ActiveDelegation` instead to be more easily separated from `UnbondingDelegation` below?